src/main/java/duke/error/ErrorChecker.java

```diff
@@ -0,0 +1,149 @@
+package duke.error;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.datetime.DateTimeConverter;
+import duke.task.Task;
+
+/**
+ * Error checker to check for errors in user input.
+ *
+ * @author  Nicole Ang
+ */
+public class ErrorChecker {
+    protected String input;
+    protected ArrayList<Task> tasks;
+    protected String errorMessage;
+
+    private boolean isDone;
+    private boolean isDelete;
+    private boolean isFind;
+    private boolean isTodo;
+    private boolean isDeadline;
+    private boolean isEvent;
+    private boolean isValidType;
+
+    /**
+     * Constructs a new ErrorChecker object used for checking errors in user input.
+     *
+     * @param input User input.
+     * @param tasks Task list.
+     */
+    public ErrorChecker(String input, ArrayList<Task> tasks) {
+        this.input = input;
+        this.tasks = tasks;
+
+        isDone = input.startsWith("done");
+        isDelete = input.startsWith("delete");
+        isFind = input.startsWith("find");
+        isTodo = input.startsWith("todo");
+        isDeadline = input.startsWith("deadline");
+        isEvent = input.startsWith("event");
+        isValidType = isDone || isDelete || isFind || isTodo || isDeadline || isEvent;
+    }
+
+    /**
+     * Returns error message.
+     *
+     * @return Error message string.
+     */
+    public String getMessage() {
+        return errorMessage;
+    }
+
+    /**
+     * Checks whether the input is valid and formatted correctly.
+     * If input is invalid, user will be prompted to re-enter valid input in correct format.
+     *
+     * @return True if input is valid and formatted correctly, and false otherwise.
+     */
+    public boolean isValid() {
+        String taskType = "";
+
+        // if (!input.equals("help") && !input.startsWith("done") && !input.startsWith("delete")
```

Comment:
I think that it would be a good idea to remove dead code here to better readability.